src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValueNew.cs

```diff
@@ -136,6 +139,36 @@ public static ColorValue New(Color value)
         public static VoidValue NewVoid()
         {
             return new VoidValue(IRContext.NotInSource(FormulaType.Void));
-        }
+        }
+
+        /// <summary>
+        /// Creates a new BlobValue from a string.
+        /// </summary>
+        /// <param name="str">String.</param>
+        /// <param name="isBase64Encoded">Is the string base64 encoded?.</param>
+        /// <param name="encoding">When the string isn't bae64 encoded, defines the encoding. Defaults to UTF8 when null or not provided.
+        /// If a base64 encoded string is provided (<paramref name="isBase64Encoded"/> is true), this parameter is unused.</param>
+        /// <returns></returns>
+        public static BlobValue NewBlob(string str, bool isBase64Encoded, Encoding encoding = null)
+        {
+            return new BlobValue(isBase64Encoded ? new Base64Blob(str) : new StringBlob(str, encoding));
+        }
+
+        public static BlobValue NewBlob(byte[] bytes)
+        {
+            return new BlobValue(new ByteArrayBlob(bytes));
+        }
+
+        /// <summary>
+        /// Creates a new BlobValue from a stream.
+        /// All content of the stream will be copied to the Blob.
+        /// The stream will not be disposed.
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <returns></returns>
+        public static BlobValue NewBlob(Stream stream)

```

Comment:
>      public static BlobValue NewBlob(Stream stream) [](http://example.com/codeflow?start=2&length=52)

Can we just remove this and add later?
Else, there's nuance to sort out like:
- when do we read the stream.
- if we call GetBytes() twice - does that double read the stream? (which may not be possible) 
- do we need to optimize out copying to bytes?

Reply:
It's all explained in the comment
- all content is copied to the blob
- we only read the bytes once